src/peer_channels.rs

```diff
@@ -480,13 +486,16 @@ pub async fn send_webxdc_realtime_data(ctx: &Context, msg_id: MsgId, data: Vec<u
 
 /// Leave the gossip of the webxdc with given [MsgId].
 pub async fn leave_webxdc_realtime(ctx: &Context, msg_id: MsgId) -> Result<()> {
-    if !ctx.get_config_bool(Config::WebxdcRealtimeEnabled).await? {
+    let Some(iroh) = ctx.get_peer_channel().await else {
+        warn!(
```

Comment:
There's no need in the warning as this is a normal case

Reply:
There's no API to know whether the peer channels are initialized. They just initialize lazily. Even if such an API appears, using it will cause race conditions as the peer channels may be initialized in parallel

Reply:
I'm not sure if I'm following. Take a look at our WebXDC preload in Desktop:

https://github.com/deltachat/deltachat-desktop/blob/1207a1a1e5d57a7a68456693240b862af794b686/packages/target-electron/static/webxdc-preload.js#L124-L141

`leave_webxdc_realtime` is supposed to be called only if the app invoked `joinRealtimeChannel`, which invokes `sendWebxdcRealtimeAdvertisement`, which is supposed to initialize `peer_channels` in Core. So a DC client knows if an app uses realtime API.